modules/coreos-layering-configuring-on.adoc

```diff
@@ -23,16 +37,41 @@ You need a separate `MachineOSConfig` CR for each machine config pool where you
+
+// Not in 4.18; maybe 4.19
+// You can manually rebuild your custom layered image by either modify your `MachineOSConfig` object or applying an annotation to the `MachineOSConfig` object as discussed in the following section.
+
+[discrete]
+[id="coreos-layering-configuring-on-limitations_{context}"]
+== On-cluster layering Technology Preview known limitations
+
+Note the following limitations when working with the on-cluster layering feature:
+
+* On-cluster layering is not supported on multi-architecture compute machines, single-node Openshift clusters, or disconnected clusters.
+* If you scale-up a machine set that uses a custom layered image, the nodes will reboot two times. The first, when the node is initially created with the base image and a second time when the custom layered image is applied.
+* Node disruption policies are not supported on nodes with a custom layered image. As a result the following configuration changes will cause a node reboot:
+** Changes to the configuration files in the `/var` or `/etc` directory
+** Adding or modifying a systemd service
+** Changing SSH keys
+** Removing mirroring rules from `ICSP`, `ITMS`, and `IDMS` objects
+** Changing the trusted CA, by updating user-ca-bundle configmap in the `openshift-config` namespace
+* The images used in creating custom layered images take up space in your push repository. Always be aware of the free space in your registry and prune the images as needed.
 
 .Prerequisites
 
-* You have enabled the `TechPreviewNoUpgrade` feature set by using the feature gates. For more information, see "Enabling features using feature gates".
+* You have enabled the `TechPreviewNoUpgrade` feature set by using the feature gates. For more information, see "Enabling features using feature gates". 
 
-* You have a copy of the global pull secret in the `openshift-machine-config-operator` namespace that the MCO needs in order to pull the base operating system image.
+* You have a copy of the pull secret in the `openshift-machine-config-operator` namespace that the MCO needs to pull the base operating system image. 
 
-* You have a copy of the `etc-pki-entitlement` secret in the `openshift-machine-api` namespace.
+// Not in 4.18; maybe in 4.19
+// If you are using the global pull secret, the MCO automatically creates a copy when you first create a `MachineOSconfig` object.
 
-* You have the push secret that the MCO needs in order to push the new custom layered image to your registry.
+* You have the push secret that the MCO needs to push the new custom layered image to your registry.
+
+* You have a copy of the `etc-pki-entitlement` secret in the `openshift-machine-api` namespace.
 
 * You have a pull secret that your nodes need to pull the new custom layered image from your registry. This should be a different secret than the one used to push the image to the repository.
```

Comment:
I don't think this is necessarily true. I know here we mention the pull and push secrets should be different https://github.com/openshift/api/blob/master/machineconfiguration/v1/types_machineosconfig.go#L71, but we don't have a field to specify the pull secret for the image built and pushed to the registry.
@cheesesashimi can you please verify? Am I missing something that changed with the API?

Reply:
I realised I was looking at the `v1` API for this. But this PR is documenting 4.18 GA, which still uses the `v1alpha1` API, so this statement is accurate and holds true.

Reply:
For v1, we assume that the pull secret is part of the global image pull secret.